admin/services/redis-cache.service.js

```diff
@@ -0,0 +1,132 @@
+ * @param redisKey - the full Redis key to get
+ * @returns {String}
+ */
+redisCacheService.get = async redisKey => {
+  if (!REDIS_CACHING) {
+    return false
+  }
+  try {
+    const result = await redis.get(redisKey)
+    if (!result) {
+      logger.info(`REDIS (get): \`${redisKey}\` is not set`)
+      return false
+    }
+    logger.info(`REDIS (get): Retrieved \`${redisKey}\``)
+    return result
+  } catch (err) {
+    logger.error(`REDIS (get): Error getting \`redisKey\`: ${err.message}`)
+    throw err
+  }
+}
+
+/**
+ * Stores data for a Redis key
+ * @param redisKey - the full Redis key to set
+ * @param data - string data to insert
+ * @returns {Boolean}
+ */
+redisCacheService.set = async (redisKey, data) => {
+  if (!REDIS_CACHING) {
+    return false
+  }
+  try {
+    if (typeof data === 'object') {
+      data = JSON.stringify(data)
+    }
+    await redis.set(redisKey, data)
+    logger.info(`REDIS (set): Stored \`${redisKey}\``)
+    return true
+  } catch (err) {
+    logger.error(`REDIS (set): Error setting \`redisKey\`: ${err.message}`)
+    throw err
+  }
+}
+
+/**
+ * Drops any supplied caches - so they can be re-queried
+ * @param caches - A single string or array of strings
+ * @returns {Boolean}
+ */
+redisCacheService.drop = async (caches = []) => {
+  if (!REDIS_CACHING || (Array.isArray(caches) && caches.length === 0)) {
+    return false
+  }
+  if (typeof caches === 'string') {
+    caches = [caches]
+  }
+  const pipeline = redis.pipeline()
+  caches.forEach(c => {
+    pipeline.del(c)
+  })
+  await pipeline.exec()
+  logger.info(`REDIS (drop): Dropped \`${caches.join('`, `')}\``)
+  return true
+}
+
+/**
+ * Manually updates data in Redis and then sends to the Azure `sql-update` queue,
+ * which will be performed in SQL Server by `/functions`
+ * @param serviceKey - the redis cache key in the format `serviceName.methodName`
+ * @param changes - object with changes in the format { tableName: 'foo-bar', update: { 1: { foo: 'bar' } } }
+ * @returns {Boolean}
+ */
+redisCacheService.update = async (serviceKey, changes) => {
+  if (!REDIS_CACHING || !REDIS_CACHE_UPDATING) {
+    return false
+  }
+  let result
+  try {
+    result = await redis.get(serviceKey)
+  } catch (err) {
+    throw err
+  }
+  result = JSON.parse(result)
+  result.recordset = result.recordset.map(r => {
+    if (changes.update && changes.update[r.id]) {
+      for (let prop in changes.update[r.id]) {
+        r[prop] = changes.update[r.id][prop]
+      }
+    }
+    if (changes.delete && changes.delete.indexOf(r.id.toString()) > -1) {
+      return false
+    }
+    return r
+  }).filter(r => r !== false)
+  try {
+    await redis.set(serviceKey, JSON.stringify(result))
+    logger.info(`REDIS (update): Updated \`${serviceKey}\``)
+    const sqlUpdateQueueName = queueNameService.getName(queueNameService.NAMES.SQL_UPDATE)
+    await azureQueueService.addMessageAsync(sqlUpdateQueueName, { version: 2, messages: [changes] })
```

Comment:
I don't advise using an SQL update queue mechanism at this time.  This is a large new part to add in just before the June trials. It would be better parked until July.

I would suggest that updates simply write to the DB first, then update the cache.

Reply:
As covered in previous discussions, this was added as a foundation for the next phase. The SQL queuing is not used at this stage. To make this clearer, all the unused areas have been removed, which does help to clarify the shape of the PR and likely reduce any confusion, so thank you for commenting on this.